wrapper/src/main/java/software/amazon/jdbc/plugin/efm/HostMonitorImpl.java

```diff
@@ -336,15 +331,15 @@ ConnectionStatus checkConnectionStatus(final long shortestFailureDetectionInterv
         LOGGER.finest(() -> "Opening a monitoring connection to " + this.hostSpec.getUrl());
         startNano = this.getCurrentTimeNano();
         // TODO: replace with ConnectionService#open
-        this.monitoringConn = this.pluginService.forceConnect(this.hostSpec, monitoringConnProperties);
-        LOGGER.finest(() -> "Opened monitoring connection: " + this.monitoringConn);
+        this.monitoringConn.set(this.pluginService.forceConnect(this.hostSpec, monitoringConnProperties));
+        LOGGER.finest(() -> "Opened monitoring connection: " + this.monitoringConn.get());
         return new ConnectionStatus(true, this.getCurrentTimeNano() - startNano);
       }
 
       startNano = this.getCurrentTimeNano();
       // Some drivers, like MySQL Connector/J, execute isValid() in a double of specified timeout time.
       // TODO: fix me. Need to find a better solution to double timeout issue.
-      final boolean isValid = this.monitoringConn.isValid(
+      final boolean isValid = this.monitoringConn.get().isValid(
```

Comment:
Similarly, do we need a local variable to guard against the case where monitoringConn.get returns null here?